src/components/common/Extensions/UserHub/partials/TransactionsTab/partials/GroupedTransaction.tsx

```diff
@@ -93,7 +94,8 @@ const GroupedTransaction: FC<GroupedTransactionProps> = ({
     values.group?.key &&
     !GROUP_KEYS_WHICH_CANNOT_LINK.includes(
       values.group.key as TRANSACTION_METHODS,
-    );
+    ) &&
+    status === TransactionStatus.Succeeded;
 
```

Comment:
Nice fix, but in terms of the single responsibility principle, wouldn’t it be better to move this check to where it’s used and just pass the prop as `isClickable={status === TransactionStatus.Succeeded}`?

Reply:
Hey @Nortsova thanks for reviewing! I actually have multiple versions of this. I had a version where I wanted to move it outside but it doesn't make sense to just move this single boolean outside, you'd have to deal with the entire lot because why stop at `status === TransactionStatus.Succeeded `? 😂
```js
const canLinkToAction =
  isClickable &&
  values.group?.key &&
  !GROUP_KEYS_WHICH_CANNOT_LINK.includes(
    values.group.key as TRANSACTION_METHODS,
  )
```
I also have another version where this component is stripped down to the point that it only has a few hooks. Would have to move around stuff outside the component etc. etc. If we really wanted to do proper SRP, you'd also have to group up every logical piece of responsibility in the component like the Header, the bit that handles the accordion-like behaviour etc.  because simply moving the boolean isn't gonna cut it. It could easily turn into a rabbithole 🐰 and I've been trying my best not to overdo it in my PRs lately 😂  Maybe someday we are free enough to audit our components and see which ones to refactor 👌